### ISSUE TITLE
partial_fit method?

### ISSUE THREAD

**Issue by dselivanov** [yixuan/tinydnn#2]
Hi Yixuan. Thanks for starting this. 
What do you think about `partial_fit` fit method (similar to `scikit-learn` API)? Since we are fitting neural network with SGD, it is natural to extend it to out-of-core computation. While `fit` will be "default" method, `partial_fit` can be more low level method - i.e. manually decrease learning rate between epochs/iterations, make some checks, etc.

And in general I'm thinking to create unified API for several types of R models (mostly inspired by scikit-learn). Mb you can be interested and can share your experience (we can discuss by email).

**Comment by yixuan** [comment]
I see. I'll take a look then.
Just to mention that `tinydnn` is more like an experimental package, since its performance is to some degrees lower than other finely tuned DNN frameworks such as MXNet. I'm not sure whether `tinydnn` will finally be published as a "formal" package or not.

**Comment by dselivanov** [comment]
But on other side for small networks it should be enough (especially for not convolutional networks). And simplicity of installation is a big advantage.

**Comment by harryprince** [comment]
it seems to migrate gluon to R is a better solution than this package.

**Comment by dselivanov** [comment]
Not sure I understand what do you mean. Better in which sense? Gluon is
python library, isn't it?

вт, 10 июл. 2018 г., 13:03 HarryZhu <notifications@github.com>:
